libs/blocks/email-collection/email-collection.js

```diff
@@ -0,0 +1,493 @@
+
+      const replace = errorMsg ? error : success;
+      if (errorMsg) window.lana?.log(errorMsg);
+
+      foreground.classList.toggle('message', !hideMessage);
+      form.classList.toggle('hidden', !hideMessage);
+      replace.classList.toggle('hidden', hideMessage);
+
+      showHideSubscribedMessage(replace, subscribed, email);
+      updateAriaLive(replace);
+
+      if (!hideMessage) return;
+      const emailEl = form.querySelector('#email');
+      if (emailEl) emailEl.value = '';
+      emailEl?.focus();
+    },
+    (children) => {
+      const foreground = children[0];
+      elsObject = {
+        foreground,
+        form: foreground.children[1],
+        success: children[1].firstElementChild,
+        error: children[2].firstElementChild,
+      };
+      return Object.values(elsObject).every((value) => value);
+    },
+  ];
+})();
+
+async function insertProgress(el, size = 'm') {
+  if (!el) return;
+
+  const { base } = miloConfig;
+  await Promise.all([
+    import('../../deps/lit-all.min.js'),
+    import(`${base}/features/spectrum-web-components/dist/theme.js`),
+    import(`${base}/features/spectrum-web-components/dist/progress-circle.js`),
+  ]);
+  let staticColor;
+  if (el.tagName === 'BUTTON') {
+    const { width, height } = window.getComputedStyle(el);
+    el.style.width = width;
+    el.style.height = height;
+    if (el.matches('.con-button.outline')) staticColor = 'white';
+  }
+  const theme = createTag(
+    'sp-theme',
+    {
+      system: 'spectrum',
+      color: 'light',
+      scale: 'medium',
+      class: 'progress-circle-container',
+    },
+  );
+  const progress = createTag(
+    'sp-progress-circle',
+    {
+      label: 'Loading content',
+      indeterminate: true,
+      'static-color': staticColor,
+      size,
+    },
+  );
+
+  theme.appendChild(progress);
+  el.replaceChildren(theme);
+}
+
+function decorateSelect(data, id, placeholder) {
+  const selectWrapper = createTag('div', { class: 'select-wrapper' });
+  const select = createTag('select', { id, name: id });
+  const optionPlaceholder = createTag('option', { value: '' }, placeholder?.trim());
+  select.appendChild(optionPlaceholder);
+
+  data.forEach(({ key, value }) => {
+    const option = createTag('option', { value: key }, value);
+    select.appendChild(option);
+  });
+
+  selectWrapper.appendChild(select);
+  return selectWrapper;
+}
+
+async function decorateInput(key, value) {
+  const profile = await getIMSProfile();
+  let inputValue = profile[key];
+  const { tag, attributes } = FORM_FIELDS[key];
+  const [labelText, placeholder] = value.split('|');
+  let input;
+
+  const label = createTag(
+    'label',
+    { for: key, class: 'body-xs' },
+    labelText.trim(),
+  );
+
+  if (key === 'country') {
+    const { country: countries } = await getFormData('config');
+    inputValue = countries?.[inputValue];
+    if (!inputValue) return null;
+  } else if (key === 'state') {
```

Comment:
rather than an else if, since the key can't be the same, it can just be an if
```suggestion
  if (key === 'country') {
    const { country: countries } = await getFormData('config');
    inputValue = countries?.[inputValue];
    if (!inputValue) return null;
  }
   if (key === 'state') {
```